src/Engines/nativeEngine.ts

```diff
@@ -1648,7 +1949,12 @@ export class NativeEngine extends Engine {
             return false;
         }
 
-        this._native.setFloatArray2(uniform, array);
+        //this._native.setFloatArray2(uniform, array);
+        this._commandBufferEncoder.beginEncodingCommand(this._native.COMMAND_SETFLOATARRAY2);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(uniform);
+        this._commandBufferEncoder.encodeCommandArgAsFloat32(array.length);

```

Comment:
Why encoding an array length with a float instead of a uint32? This scares me as soon as the length is getting big.

Reply:
Good eye @CedricGuillemet! This was actually an error. It is encoded into the float buffer and decoded from the uint buffer (which doesn't work). I already have a fix for this locally, but haven't pushed those changes yet. I will probably do so today.